Documentation/ApiOverview/DataHandler/Introduction/Index.rst

```diff
@@ -1,31 +1,32 @@
-.. include:: /Includes.rst.txt
-.. _tce-introduction:
+..  include:: /Includes.rst.txt
+..  _tce-introduction:
+..  _datahandler-introduction:
 
 ============
 Introduction
 ============
 
-
-.. index::
-   pair: DataHandler; Database
-   pair: DataHandler; TCA
-.. _tce-database:
+..  index::
+    pair: DataHandler; Database
+    pair: DataHandler; TCA
+..  _tce-database:
+..  _datahandler-database:
 
 Database
 ========
 
-The TYPO3 Core engine is the class that handles *all* ***data***
-*writing to database tables configured in $TCA*. In addition the
-class handles **commands** such as copy, move, delete. It will handle
+The DataHandler is the class that handles **all** :ref:`data <tce-data>`
+writing to database tables configured in TCA. In addition the class handles
+:ref:`commands <tce-commands>` such as copy, move, delete. It will handle
 undo/history and versioning of records and everything will be logged
-to the sys\_log. And it will make sure that write permissions are
-evaluated correctly for the user trying to write to the database.
-Generally, any processing specific option in the $TCA array is handled
+to the `sys_log` table. It will make sure that write permissions are
```

Comment:
```suggestion
to the :sql:`sys_log` table. It will make sure that write permissions are
```

Reply:
Nope, that wa sby intention. It has no visual impact anymore. So, I won't do that.

Reply:
Let me explain, why I don't think it is worth to use those roles.

First, I see a misconception in guides itself: in Sphinx an undefined role throws a warning, so you can spot an error/typo easily, for example `sgl` instead of `sql` (as they must be defined beforehand). In guides, every role is accepted.

Second, as we have no warning, we may have a visual display of a role, like a color (not preferred) or a text/symbol  before/after., so a writer gets a grasp, there is something wrong, and an editor knows at a glance what is that thing there.
As we don't have neither a warning nor a visual representation of that role in guides/our render-guides, you can't really spot typos in roles. So what you write there, `sql`, `php`, `fuck` or something else, it does not really matter.

Third, it is superfluous: it creates overhead on writing, and it creates overhead on reviewing. It is not worth the work as it has no value to use roles. Therefore, I will omit roles where they have no real benefit (for me as writer and for the reader).

Reply:
@brotkrueml, you have a point there about non-existant roles not throwing warnings. Would you like to make an issue about that? Also for displaying the code-language we are still working on solutions that should serve everyone. At least they should be easier to understand then the former color-code.

Reply:
But please let us move this topic into issues where it belongs and not unrelated reviews.

Reply:
See: https://github.com/phpDocumentor/guides/issues/925